### PR TITLE
Adds cancel button to jobs list

### PR DESCRIPTION
**Pull request by mabashian**
##### SUMMARY
link #7364 

Screen Shot 2020-08-25 at 1 54 32 PM

canceljob

This feature adds a cancel button next to the delete button on the jobs list.  This cancel button behaves like the delete button in that a user can select multiple jobs to cancel at one time.  Some general rules surrounding the button:

1) Button should be disabled until a user selects at least one job
2) Button should be disabled if user selects a job that they don't have the ability to cancel (summary_fields.user_capabilities.start is false)
3) API requests should not be made for jobs that are in a finished state.  A user can still select these rows but confirming a cancel will not actually send the request(s).  I confirmed this behavior with Taufique and Bill.

I also wanted to use the `copy` related endpoint provided on all of the job objects rather than having a big `switch` statement covering all of the different job types.  Since we didn't have anything that would support taking in an arbitrary endpoint I created `Related.js` in our models directory.  This model has generic methods covering GET/POST/DELETE/PUT/PATCH.  It doesn't inherit from the base model so I needed to pull the axios http instantiation out into something that could be consumed by more than one model.

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
 - UI

### CODE REVIEW

File: awx/ui_next/src/components/JobList/JobListCancelButton.jsx [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+  const renderTooltip = () => {
+    const jobsUnableToCancel = jobsToCancel
+      .filter(cannotCancel)
+      .map(job => job.name)
+      .join(', ');
+    if (jobsToCancel.some(cannotCancel)) {
+      return (
+        <div>
+          {i18n.plural({
```
Comment (mabashian) [inline awx/ui_next/src/components/JobList/JobListCancelButton.jsx]: i18n.plural is no good -> see #8002 

Use low level i18n._ instead

**Review by jakemcdermott — CHANGES_REQUESTED** [review]
Looking good & works well. We dove into job-switching problem offline. Ultimately @mabashian and I decided on adding a util to select the job model given a `job` object as a way to avoid having to add a switch everywhere a job api is used.

**Review by jakemcdermott — APPROVED** [review]

**Comment by unlikelyzero** [comment]
Tiago will be working on this one today

**Comment by tiagodread** [comment]
@mabashian when I tried to cancel a running job using the Cancel button inside kebab (when the advanced search is enabled) nothing happens (the same occurred for delete button)

cancel_button

**Comment by tiagodread** [comment]
recheck

**Review by tiagodread — APPROVED** [review]
Tests and stubs in place 👍🏽

**Comment by tiagodread** [comment]
regate

**Pull request merged**